review/views.py

```diff
@@ -350,3 +359,222 @@ def exercise_settings(request, course_key=None, exercise_key=None, course=None,
         "name": ''
     })
     return render(request, "review/exercise_settings.html", context)
+
+
+@access_resource
+def students_view(request, course=None, course_key=None):
+    
+    """ This view too slow without cache..."""
+
+    # Comparisons for a course (excluding template comparisons)
+    comparisons = Comparison.objects\
+        .filter(submission_a__exercise__course=course)\
+        .filter(similarity__gt=0)\
+        .exclude(submission_b__isnull=True)
+
+    submissions = Submission.objects\
+        .filter(exercise__course=course)\
+
+    students = course.students.all()
+
+    # For average similarity score
+    averages = {}
+
+    # For highest similarity scores
+    highest = {}
+    highest_with = {}
+
+    # Create average scores for all students in the course
+    for student in students:
+
+            # Comparisons where student is either a or b
+            similarities = (
+                comparisons\
+                    .filter(Q(submission_a__student=student) | Q(submission_b__student=student))\
```

Comment:
There may be hundreds of students in the course. Usually, repeating a database query in a loop like this becomes too slow and loading the page would take too long. It is worthwhile to try to move these queries to a large batch query before the loop so that there are no queries in the loop. A single query would fetch data for all students and you would group them somehow so that you can iterate over the students (in a loop).

Reply:
Yes, this is the problem. I'll try this suggestion out.